File: Task 2/Java/app/src/main/java/dev/ogabek/java/receiver/CallBroadcastReceiver.java
package dev.ogabek.java.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.telephony.TelephonyManager;
import android.view.Gravity;
import android.widget.Toast;

public class CallBroadcastReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getStringExtra(TelephonyManager.EXTRA_STATE);
        if (action.equals(TelephonyManager.EXTRA_STATE_OFFHOOK)) {
            showToast(context, "Call started...");
        } else if (action.equals(TelephonyManager.EXTRA_STATE_IDLE)) {
            showToast(context, "Call ended...");
        } else if (action.equals(TelephonyManager.EXTRA_STATE_RINGING)) {
            showToast(context, "Incoming call...");
        }
    }

    private void showToast(Context context, String message) {
        Toast toast = Toast.makeText(context, message, Toast.LENGTH_LONG);
        toast.setGravity(Gravity.CENTER, 0, 0);
        toast.show();
    }

}

File: Task 2/Kotlin/app/src/main/java/dev/ogabek/kotlin/receiver/CallBroadcastReceiver.kt
package dev.ogabek.kotlin.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.telephony.TelephonyManager
import android.widget.Toast
import android.view.Gravity


class CallBroadcastReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        when {
            intent.getStringExtra(TelephonyManager.EXTRA_STATE) == TelephonyManager.EXTRA_STATE_OFFHOOK -> {
                showToast(context, "Call started...")
            }
            intent.getStringExtra(TelephonyManager.EXTRA_STATE) == TelephonyManager.EXTRA_STATE_IDLE -> {
                showToast(context, "Call ended...")
            }
            intent.getStringExtra(TelephonyManager.EXTRA_STATE) == TelephonyManager.EXTRA_STATE_RINGING -> {
                showToast(context, "Incoming call...")
            }
        }
    }

    private fun showToast(context: Context?, message: String?) {
        val toast = Toast.makeText(context, message, Toast.LENGTH_LONG)
        toast.setGravity(Gravity.CENTER, 0, 0)
        toast.show()
    }
}